src/wallet/wallet.cpp

```diff
@@ -2326,16 +2326,18 @@ void CWallet::ClearTxns(const std::vector<uint256>& tx_hashes)
 DBErrors CWallet::ZapSelectTx(std::vector<uint256>& vHashIn)
 {
     AssertLockHeld(cs_wallet);
-    DBErrors nZapSelectTxRet = WalletBatch(GetDatabase()).ZapSelectTx(vHashIn);
-    // Remove transactions from the memory map
-    ClearTxns(vHashIn);
 
-    if (nZapSelectTxRet != DBErrors::LOAD_OK)
-        return nZapSelectTxRet;
+    WalletBatch batch(GetDatabase());
+    if (!batch.TxnBegin()) return DBErrors::NONCRITICAL_ERROR;
+    DBErrors db_status = batch.ZapSelectTx(vHashIn);
+    // All deletions should have been added to the db txn, otherwise we abort the process.
+    if (db_status != DBErrors::LOAD_OK) return db_status;
```

Comment:
In d26a28508d5322fe7c29b3642dcf3a17b0ea31f1 "wallet: batch ZapSelectTx db operations"

This should explicitly call `TxnAbort` before returning as otherwise we will get extraenous logs about the tx being aborted before batch is destroyed.

Reply:
> This should explicitly call TxnAbort before returning as otherwise we will get extraenous logs about the tx being aborted before batch is destroyed.

sure. Fixed.